rotkehlchen/api/rest.py

```diff
@@ -4125,3 +4120,111 @@ def get_all_counterparties(self) -> Response:
                 'message': '',
             },
         )
+
+    def get_addressbook_entries(
+            self,
+            book_type: AddressbookType,
+            addresses: Optional[List[ChecksumEthAddress]],
+    ) -> Response:
+        db_addressbook = DBAddressbook(self.rotkehlchen.data.db)
+        entries = db_addressbook.get_addressbook_entries(
+            book_type=book_type,
+            addresses=addresses,
+        )
+        serialized = [entry.serialize() for entry in entries]
+        return api_response(_wrap_in_ok_result(serialized))
+
+    def add_addressbook_entries(
+            self,
+            book_type: AddressbookType,
+            entries: List[AddressbookEntry],
+    ) -> Response:
+        db_addressbook = DBAddressbook(self.rotkehlchen.data.db)
+        try:
+            db_addressbook.add_addressbook_entries(book_type=book_type, entries=entries)
+            return api_response(result=OK_RESULT)
+        except InputError as e:
+            return api_response(
+                result=wrap_in_fail_result(str(e)),
+                status_code=HTTPStatus.CONFLICT,
+            )
+
+    def update_addressbook_entries(
+            self,
+            book_type: AddressbookType,
+            entries: List[AddressbookEntry],
+    ) -> Response:
+        db_addressbook = DBAddressbook(self.rotkehlchen.data.db)
+        try:
+            db_addressbook.update_addressbook_entries(book_type=book_type, entries=entries)
+            return api_response(result=OK_RESULT)
+        except InputError as e:
+            return api_response(
+                result=wrap_in_fail_result(str(e)),
+                status_code=HTTPStatus.CONFLICT,
+            )
+
+    def delete_addressbook_entries(
+            self,
+            book_type: AddressbookType,
+            addresses: List[ChecksumEthAddress],
+    ) -> Response:
+        db_addressbook = DBAddressbook(self.rotkehlchen.data.db)
+        try:
+            db_addressbook.delete_addressbook_entries(book_type=book_type, addresses=addresses)
+            return api_response(result=OK_RESULT)
+        except InputError as e:
+            return api_response(
+                result=wrap_in_fail_result(str(e)),
+                status_code=HTTPStatus.CONFLICT,
+            )
+
+    def search_for_names_everywhere(self, addresses: List[ChecksumEthAddress]) -> Response:
+        db_addressbook = DBAddressbook(self.rotkehlchen.data.db)
+        db_ens = DBEns(self.rotkehlchen.data.db)
+
+        reverse_ens = db_ens.get_reverse_ens(addresses=addresses).values()
+        tokens = {}
+        for address in addresses:
+            token = GlobalDBHandler().get_ethereum_token(address=address)
+            if token is not None:
+                tokens[token.ethereum_address] = token.name
```

Comment:
Yes, makes sense. I also thought about it but didn't want to add a new method for this particular use-case. If you also think that it would be a good enhancement, I'll implement it then

Reply:
Done